packages/utils/telemetry-utils/src/mockLogger.ts

```diff
@@ -141,7 +141,22 @@ ${JSON.stringify(actualEvents)}`);
 		actual: ITelemetryBaseEvent,
 		expected: Omit<ITelemetryBaseEvent, "category">,
 	): boolean {
-		const masked = { ...actual, ...expected };
-		return JSON.stringify(masked) === JSON.stringify(actual);
+		// "details" is used in a lot of telemetry logs to group a bunch of properties together and stringify them.
+		// Some of the properties in the expected event may be inside "details". So, if the expected event does not have
+		// "details", extract the properties from "details" in the actual event and inline them in the actual event.
+		const { details, ...actualForMatching } = actual;
+		let detailsExpanded = { details };
+		if (
+			expected.details === undefined &&
```

Comment:
This all feels too creepy and implicit.  How about adding a default-false parameter `inlineDetailsProp` and do this if true (probably ignore `expected.details`, or even throw if that is there and would be overwritten based on the param)

Reply:
Please also write a few UTs for this (mockLogger.spec.ts)